firedrake/utility_meshes.py

```diff
@@ -1360,25 +1336,27 @@ def PartiallyPeriodicRectangleMesh(nx, ny, Lx, Ly, direction="x", quadrilateral=
     # make x-periodic mesh
     # unravel x coordinates like in periodic interval
     # set y coordinates to z coordinates
-    periodic_kernel = """double Y,pi;
-            Y = 0.0;
-            for(int i=0; i<old_coords.dofs; i++) {
-                Y += old_coords[i][1];
-            }
-
-            pi=3.141592653589793;
-            for(int i=0;i<new_coords.dofs;i++){
-            new_coords[i][0] = atan2(old_coords[i][1],old_coords[i][0])/pi/2;
-            if(new_coords[i][0]<0.) new_coords[i][0] += 1;
-            if(new_coords[i][0]==0 && Y<0.) new_coords[i][0] = 1.0;
-            new_coords[i][0] *= Lx[0];
-            new_coords[i][1] = old_coords[i][2]*Ly[0];
-            }"""
+
+    domain = "{[i, j]: 0 <= i < old_coords.dofs and 0 <= j < new_coords.dofs}"
+    instructions = """
+    <float64> Y = 0
+    <float64> pi = 3.141592653589793
+    for i
+        Y = Y + old_coords[i, 1]
+    end
```

Comment:
Can we write this as the reduction it is? `<float64> Y = sum(i, old_coords[i, 1])`?

Reply:
(Minor issue/comment)

Reply:
Also, I was thinking the reduction in the tsfc generated assembly kernels could be written in similar ways. Maybe one step at a time?